src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const fromParts = fromFormat.slice(0, 3);
+  const toParts = toFormat.slice(0, 3);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  for (let i = 0; i < fromParts.length; i++) {
+    dateMap[fromParts[i]] = dateParts[i];
+  }
+
+  if (fromParts.includes('YYYY') && toParts.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (fromParts.includes('YY') && toParts.includes('YYYY')) {
```

Comment:
Consider refactoring the conditional blocks that handle year conversion into a separate function. This will make the `formatDate` function more readable and maintainable by abstracting the year conversion logic. You can create a function like `convertYear` that takes the `dateMap` and the `fromParts`, `toParts` as arguments and performs the necessary conversion.